mesonbuild/coredata.py

```diff
@@ -207,8 +206,8 @@ def __post_init__(self, value: T.Union[str, OctalInt]) -> None:
         self.choices: T.List[str] = ['preserve', '0000-0777']
         super().__post_init__(value)
 
-    def printable_value(self) -> str:
-        if self.value == 'preserve':
+    def printable_value(self) -> StringProtocol:
+        if isinstance(self.value, str):
             return self.value
         return format(self.value, '04o')
```

Comment:
yeah, because we haven't actually annotated value to be `Literal['presever'] | int`. Maybe we can try that now, actually...

Reply:
I tried this, but ended up breaking mypy somehow. Needs more investigating...

Reply:
This is kinda challenging, because we expect to be able to pass in string values representing octals like `'022'` in `install_umask`. I don't think we can have our cake and eat it too.